### PR TITLE
feat(artifacts): take appNotReady.png screenshot if needed

### PR DESCRIPTION
**Pull request by noomorph**
- [x] This is a small change 

---

**Description:**

Supplements #1914 . Will be taking an automatic screenshot `appNotReady.png`, when there was an issue with launching the app, when it never signalled back that it is ready and active.

@d4vidi, here the question is mostly whether `device.driver.emitter` makes you smell _Feature envy_, one of the code smells, or it is fine?

### CODE REVIEW

File: detox/src/devices/Device.js [hunk 1]
```diff
@@ -86,6 +86,11 @@ class Device {
 
     await this.deviceDriver.waitUntilReady();
     await this.deviceDriver.waitForActive();
+    await this.deviceDriver.emitter.emit('appReady', {
```
Comment (d4vidi) [inline detox/src/devices/Device.js]: why not tested in device unit tests?

**Comment by d4vidi** [comment]
>here the question is mostly whether device.driver.emitter makes you smell Feature envy
😄 
TBH it's weird for me to begin with that the emitter is a strict property of the driver. Feels like it should be an object shared among the designated emitters. You know what I mean?

File: detox/src/devices/Device.js (hunk 1 above)
Comment (noomorph) [reply detox/src/devices/Device.js]: added a test, thanks.

**Comment by noomorph** [comment]
I have rewritten emitters code according to what you suggested earlier, so, now we have a single emitter instance per one Detox, and it is passed down to devices and device drivers.

**Pull request merged**